specification.md

```diff
@@ -485,8 +485,8 @@ into. With this, after our action executes the state data would be:
 
 | Parameter | Description | Type | Required |
 | --- | --- | --- | --- |
-| data | Workflow expression that filters the event data (payload) | string | no |
-| toStateData | Workflow expression that selects a state data element to which the action results should be added/merged into. If not specified denotes the top-level state data element | string | no |
+| data | Workflow expression that filters the event data (payload) | expression | no |
+| toStateData | Workflow expression that selects a state data element to which the action results should be added/merged into. If not specified denotes the top-level state data element | path expression | no |
```

Comment:
please define what is a path expression in its own section so its clear and link to it here

Reply:
that is mainly for discussion because we're not explaining that the expression used in toStateData is auto-completed by a runtime, e.g. `${ .path.to.element }` will be executed as
`jq ' .path.to.element  = $result' --argjson result '{...}' /statedatafile`
I agree we need to be clear about it. If, however, we want to pass the result to the expression, e.g. through $action.result, we could avoid this `path expression` type and make the default expression `${ $action.result }`

Reply:
removed `path expression` in favor of not having pointer-like fields in the spec.